File: Client/src/app/dtos/responses/order.dto.ts
import { DeliveryMethodDto } from "./delivery-method.dto"
import { OrderItemDto } from "./order-item.dto"

export interface OrderDto {
    id: number
    orderDate: string
    deliveryMethod: DeliveryMethodDto
    items: OrderItemDto[]
    status: string
    subTotal: number
    total: number
}

File: WebApi/Dtos/Respons/OrderDto.cs

namespace WebApi.Dtos.Respons
{
    public class OrderDto
    {
        public int Id { get; set; }
        public DeliveryMethodDto DeliveryMethod { get; set; }
        public DateTime OrderDate { get; set; }
        public ICollection<OrderItemDto> Items { get; set; }
        public string Status { get; set; }
        public decimal SubTotal { get; set; }
        public decimal Total { get; set; }
    }
}
